tests/lzma.rs

```diff
@@ -75,30 +100,79 @@ fn round_trip_file(filename: &str) {
 fn decomp_big_file(compfile: &str, plainfile: &str) {
     use std::io::Read;
 
-    let mut expected = Vec::new();
-    std::fs::File::open(plainfile)
-        .unwrap()
-        .read_to_end(&mut expected)
-        .unwrap();
-    let mut f = std::io::BufReader::new(std::fs::File::open(compfile).unwrap());
-    let mut decomp: Vec<u8> = Vec::new();
-    lzma_rs::lzma_decompress(&mut f, &mut decomp).unwrap();
-    assert!(decomp == expected)
+    let expected = {
+        let mut expected = Vec::new();
+        std::fs::File::open(plainfile)
+            .unwrap()
+            .read_to_end(&mut expected)
+            .unwrap();
+        expected
+    };
+
+    // test non-streaming decompression
+    {
+        let input = {
+            let mut input = Vec::new();
+            std::fs::File::open(compfile)
+                .unwrap()
+                .read_to_end(&mut input)
+                .unwrap();
+            input
```

Comment:
Since this pattern appears twice already in this function, it would make sense to create a utility function for it.